PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java

```diff
@@ -313,6 +307,31 @@ protected void createUIComponents() {
         yarnEndpointField = new HintTextField("(Optional)Example: http://hn0-spark2:8088");
         arisClusterNameField = new HintTextField("(Optional) Cluster name");
         arisHostField = new HintTextField("Example: 10.123.123.123");
+
+        clusterNameOrUrlField.addFocusListener(new FocusAdapter() {
+            @Override
+            public void focusLost(FocusEvent e) {
+                String clusterNameOrUrl = clusterNameOrUrlField.getText();
+
+                if (StringUtils.isBlank(clusterNameOrUrl)) {
+                    return;
+                }
+
+                try {
+                    AuthType authType = SparkBatchSubmission.getInstance().probeAuthType(
+                            getClusterConnectionUrl(clusterNameOrUrl.trim()));
+
+                    authComboBox.setSelectedItem(authType);
```

Comment:
What if the authType is `AuthType.NotSupported`

Reply:
In my test, nothing will happen if the type isn't matched.